src/wovnio/wovnphp/core/tests/unit/WovnURLTest.php

```diff
@@ -0,0 +1,262 @@
+    }
+
+    public function testConstructorWithPathPatternDefaultLangWithPathPrefix()
+    {
+        $directory = $this->getLangDirectory();
+        $options = $this->getOptions(array(WovnOption::OPT_URL_PATTERN_NAME => 'path', WovnOption::OPT_SITE_PREFIX_PATH => 'news/blog'));
+        $originalUrl = 'https://www.example.com/news/blog/';
+        $url = new WovnURL($originalUrl, $directory, $options);
+        self::assertEquals('en', $url->lang()->code());
+    }
+
+    public function testConstructorWithPathPatternDefaultLangWithPathPrefixVariation1()
+    {
+        $directory = $this->getLangDirectory();
+        $options = $this->getOptions(array(WovnOption::OPT_URL_PATTERN_NAME => 'path', WovnOption::OPT_SITE_PREFIX_PATH => 'news/blog'));
+        $originalUrl = 'https://www.example.com/news/blog/ja';
+        $url = new WovnURL($originalUrl, $directory, $options);
+        self::assertEquals('en', $url->lang()->code());
+    }
+
+    public function testConstructorWithPathPatternDefaultLangWithPathPrefixVariation2()
+    {
+        $directory = $this->getLangDirectory();
+        $options = $this->getOptions(array(WovnOption::OPT_URL_PATTERN_NAME => 'path', WovnOption::OPT_SITE_PREFIX_PATH => 'news/blog'));
+        $originalUrl = 'https://www.example.com/ja/news/blog/ja';
+        $url = new WovnURL($originalUrl, $directory, $options);
+        self::assertEquals('en', $url->lang()->code());
+    }
+
+    public function testConstructorWithPathPatternNonDefaultLangWithPathPrefix()
+    {
+        $directory = $this->getLangDirectory();
+        $options = $this->getOptions(array(WovnOption::OPT_URL_PATTERN_NAME => 'path', WovnOption::OPT_SITE_PREFIX_PATH => 'news/blog'));
+        $originalUrl = 'https://www.example.com/news/blog/ja/';
+        $url = new WovnURL($originalUrl, $directory, $options);
+        self::assertEquals('ja', $url->lang()->code());
+    }
+
+    public function testConstructorWithPathPatternNonDefaultLangWithPathPrefixVariation1()
+    {
+        $directory = $this->getLangDirectory();
+        $options = $this->getOptions(array(WovnOption::OPT_URL_PATTERN_NAME => 'path', WovnOption::OPT_SITE_PREFIX_PATH => 'news/blog'));
+        $originalUrl = 'https://www.example.com/news/blog/ja/page.html';
+        $url = new WovnURL($originalUrl, $directory, $options);
+        self::assertEquals('ja', $url->lang()->code());
+    }
+
+    public function testConstructorWithPathPatternNonDefaultLangWithPathPrefixVariation2()
+    {
+        $directory = $this->getLangDirectory();
+        $options = $this->getOptions(array(WovnOption::OPT_URL_PATTERN_NAME => 'path', WovnOption::OPT_SITE_PREFIX_PATH => 'news/blog'));
+        $originalUrl = 'https://www.example.com/news/blog/ja/text/next/page.html';
+        $url = new WovnURL($originalUrl, $directory, $options);
+        self::assertEquals('ja', $url->lang()->code());
+    }
+
+    public function testConstructorWithPathPatternNonDefaultLangWithPathPrefixVariation3()
```

Comment:
For these test names, i'm seeing `Constructor` a lot but what you are actually testing is the `lang` function.